src/memex/models/document.py

```diff
@@ -463,13 +487,19 @@ def update_document_metadata(session,
 
     document.updated = updated
 
+    document_uris = []
     for document_uri_dict in document_uri_dicts:
-        create_or_update_document_uri(
-            session=session,
-            document=document,
-            created=created,
-            updated=updated,
-            **document_uri_dict)
+        document_uris.append(
+            create_or_update_document_uri(
+                session=session,
+                document=document,
+                created=created,
+                updated=updated,
+                **document_uri_dict)
+        )
+
+    if not document.web_uri:
```

Comment:
I know this is consistent with the previous code, but shouldn't this denormalised field be updated every time the set of URIs is updated?